python/xorbits/_mars/lib/filesystem/_glob.py

```diff
@@ -157,11 +157,14 @@ def _glob2(self, dirname, pattern, dironly):  # pragma: no cover
     def _iterdir(self, dirname, dironly):
         if not dirname:  # pragma: no cover
             dirname = ""
-        if not self._fs.isdir(dirname):
+        if self._fs.isfile(dirname):
+            return iter([])
+        try:  # pragma: no cover
```

Comment:
Could you explain a little bit about the try and except? Like when will an exception be raised?

Reply:
when the dirname does not exist in the fs, it will just return an empty iterator.